File: src/algorithms/find-duplicates.js
/**
 * Given an integer array nums where each integer appears at most twice,
 * returns an array of all integers that appear twice.
 * 
 * @param {number[]} nums - An array of positive numbers
 * @returns {number[]} - Returns an array of numbers that appear twice in nums
 */
function findDuplicates (nums) {
  let res = [];
  let elems = new Set();
  for (const num of nums) {
    if (!elems.has(num)) {
      elems.add(num);
    } else {
      res.push(num);
    }
  }
  return res;
}

File: src/algorithms/find-duplicates.py
def findDuplicates(nums):
    """Given an integer array nums of length n where all the integers of nums 
    are in the range [1, n] and each integer appears at most twice, return an 
    array of all the integers that appears twice.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        list[int]: Returns a list of integers that appear twice in nums
    """
    res = []
    seen = set()
    for num in nums:
        if num in seen:
            res.append(num)
        else:
            seen.add(num)
    return res
